tests/Rules.Framework.Tests/Evaluation/Compiled/RuleConditionsExpressionBuilderTests.GoldenFile1.csx

```diff
@@ -0,0 +1,75 @@
+private Func<object, Operators, object, string> Evaluate1;
+private Func<IDictionary<ConditionType, object>, ConditionType, object> GetValueOrDefault1;
+
+internal bool Main(EvaluationContext<ConditionType> evaluationContext)
+{
+    bool _C0_result;
```

Comment:
I refuse to review this code 😆 

João `goto ` end

Reply:
Yes, it is awful I know, but I can explain the logic behind those variable names.

The "C" stands for "condition" and the number after is an ordered number to distinguish the conditions inside a `ComposedConditionNode`. Let's suppose you have a composed condition, using the operator And, and having 2 child conditions: you'll have variables starting with _C0_* for the first condition and variables starting with _C1_* for the second condition. I did this to avoid enormous variable names, because the bigger the conditions tree, the bigger the name of the variables.

Reply:
Yes, I was joking and not really protesting 😛 But the rationale you've presented makes sense.

Since we're discussing this, I'd say that personally I'm not a fan of using underscore for private fields.. 
I've tried to understand what are Microsoft's recommendations, and although it seems that the usage of "_" is kind of a legacy of C/C++, it is not really incentivized (nor unincentivized) by Microsoft :
"_" appears on their examples in [Fields documentation](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/fields) , but it doesn't on their [Coding conventions](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions).

Reply:
I can look up on this and see if I can remove the underscores and use another pattern for this 👌 I don't mind doing that. It was done this way because the compilation results were not meant to be read by a developer (on the Rules.Framework user perspective), but we have to account for the Rules.Framework developers (ourselves) 😛

Reply:
Yes, it's nit-picking but we do have to make life easier for us and others 😛

Reply:
I've now reviewed the logic to improve names. Only one variable name is not following the convention, the "Result". But in order to get that one to follow the conventions, I'd have to make substantial changes to the codebase, and I think it is not worth the effort. Can you take a look? I think it is a lot more readable 👌